packages/mui-material/src/Modal/useModal.ts

```diff
@@ -83,7 +84,13 @@ function useModal(parameters: UseModalParameters): UseModalReturnValue {
   };
 
   const handleOpen = useEventCallback(() => {
-    const resolvedContainer = getContainer(container) || getDoc().body;
+    /**
+     * Resolving this could be simplified (mountNodeRef should take priority when it's set)
+     * but this will also work because the logic matches {@link Portal}.
+     */
+    const resolvedContainer = disablePortal
+      ? ((mountNodeRef.current ?? modalRef.current)?.parentElement ?? getDoc().body)
```

Comment:
When `disablePortal` is true, `mountNodeRef` and `modalRef` are the same. So why should `mountNodeRef` have priority?

Reply:
naming-wise `mountNodeRef` is the node `modalRef` is mounted to, so it makes sense even if they are the same.

Plus after reading the `Portal` implementation I can see that `mountNodeRef` can still come back as `null` even if the typing for it is not indicating it.